rustler/src/alloc.rs

```diff
@@ -0,0 +1,20 @@
+use std::alloc::{GlobalAlloc, Layout};
+
+#[cfg(feature = "allocator")]
+#[global_allocator]
+static ALLOCATOR: EnifAllocator = EnifAllocator;
+
+/// Allocator implementation that forwards all allocation calls to Erlang's allocator. Allows the
+/// memory usage to be tracked by the BEAM.
+pub struct EnifAllocator;
+
+unsafe impl GlobalAlloc for EnifAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        // TODO: Check the requested alignment.
+        rustler_sys::enif_alloc(layout.size()) as *mut u8
```

Comment:
i couldn't reproduce any memory issues on my linux machine, but i did notice that this may need to be something like `layout.size.max(layout.align())`?

Reply:
Good point, let me try that.

Reply:
Nope, that's not it.